subt_ign/src/GameLogicPlugin.cc

```diff
@@ -2307,6 +2317,17 @@ void GameLogicPluginPrivate::Finish(const ignition::msgs::Time &_simTime)
       completeMsg.set_data("recording_complete");
       this->startPub.Publish(completeMsg);
 
+      std::lock_guard<std::mutex> lock(this->eventCounterMutex);
+      std::ostringstream stream;
+      stream
+        << "- event:\n"
+        << "  id: " << this->eventCounter << "\n"
+        << "  type: recording_complete\n"
+        << "  time_sec: " << _simTime.sec() << "\n"
+        << "  elapsed_real_time: " << realElapsed << "\n"
+        << "  elapsed_sim_time: " << simElapsed << "\n"
+        << "  total_score: " << this->totalScore << std::endl;
+      this->LogEvent(stream.str());
       this->eventCounter++;
     }
```

Comment:
Could we add an `else` case to mention that the ROS node is not active? That way we can verify that the block has not been accessed.

Reply:
the `if (this->rosnode)` block ends on line 2310. The new event starting on line 2320 will always output, regardless of ROS usage. That event will be logged in `events.yml`.

Reply:
Ah, my bad. My eyes are used to larger indents, or maybe I need some glasses. Thanks!

Reply:
Ah, my bad. My eyes are used to larger indents, or maybe I need glasses. Thanks!